org.kde.francis.json

```diff
@@ -0,0 +1,61 @@
+{
+    "id": "org.kde.francis",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15-22.08",
+    "sdk": "org.kde.Sdk",
+    "command": "francis",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri"
+    ],
+    "separate-locales": false,
+
+    "modules": [
+        {
+            "name": "kirigami-addons",
+            "config-opts": [
+                "-DBUILD_TESTING=OFF",
+                "-DCMAKE_BUILD_TYPE=Release"
+            ],
+            "buildsystem": "cmake-ninja",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/kirigami-addons/kirigami-addons-0.10.0.tar.xz",
+                    "sha256": "9897e70bf167f579cec06d09d49a9a27aba184cbd1b2fce2c3ae757a87b01216",
+                    "x-checker-data": {
+                        "type": "anitya",
+                        "project-id": 242933,
+                        "stable-only": true,
+                        "url-template": "https://download.kde.org/stable/kirigami-addons/kirigami-addons-$version.tar.xz"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "francis",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DBUILD_TESTING=OFF",
+                "-DCMAKE_BUILD_TYPE=Release"
+            ],
+            "builddir": true,
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/francis/francis-1.0.1.tar.xz",
```

Comment:
The first and last one needs to be fixed.

```
appstream-util validate org.kde.francis.appdata.xml 
org.kde.francis.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : <image> has vertical padding [https://cdn.kde.org/screenshots/francis/main.png]
• style-invalid         : <image> has horizontal padding [https://cdn.kde.org/screenshots/francis/main.png]
• style-invalid         : <caption> is too short [Main View];shortest allowed is 10 chars
Validation of files failed
```

Reply:
```
desktop-file-validate org.kde.francis.desktop 
org.kde.francis.desktop: error: file contains key "SingleMainWindow" in group "Desktop Entry", but keys extending the format should start with "X-"
```

desktop-file-validate from Freedesktop SDK is missing this. This is not included in a desktop-file-utils release, distributions patch it

Reply:
> appstream runs desktop file validation, so it'll fail I think.

Looks like it got reduced to a warning, so it should be fine

```
WARNING: Failed to validate desktop file /srv/buildbot/worker/build-x86_64-3/build/builddir/export/share/applications/org.kde.francis.desktop: /srv/buildbot/worker/build-x86_64-3/build/builddir/export/share/applications/org.kde.francis.desktop: error: file contains key "SingleMainWindow" in group "Desktop Entry", but keys extending the format should start with "X-"
```

Reply:
> • style-invalid         : <image> has vertical padding [https://cdn.kde.org/screenshots/francis/main.png]
> • style-invalid         : <image> has horizontal padding [https://cdn.kde.org/screenshots/francis/main.png]

Looks like the checker doesn't like the screenshot taken with the official guidelines for taking screenshot in KDE :( Fortunately this is just a warning

Reply:
The check that buildbot use doesn't even emit a warning for this.